ert/data/record/_transformation.py

```diff
@@ -22,71 +24,80 @@ def _prepare_location(base_path: Path, location: Path) -> None:
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
         raise NotImplementedError("not implemented")
 
     @abstractmethod
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
+    async def transform_output(self, mime: str, location: Path) -> Record:
         raise NotImplementedError("not implemented")
 
 
 class FileRecordTransformation(RecordTransformation):
     async def transform_input(
-        self,
-        transmitter: RecordTransmitter,
-        mime: str,
-        runpath: Path,
-        location: Path,
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
         _prepare_location(runpath, location)
-        await transmitter.dump(runpath / location, mime)
+        await _save_record_to_file(record, runpath / location, mime)
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        await transmitter.transmit_file(runpath / location, mime)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return await _load_record_from_file(location, mime)
 
 
 class TarRecordTransformation(RecordTransformation):
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
-        record = await transmitter.load()
         if isinstance(record, BlobRecord):
             with tarfile.open(fileobj=io.BytesIO(record.data), mode="r") as tar:
                 _prepare_location(runpath, location)
                 tar.extractall(runpath / location)
         else:
             raise TypeError("Record needs to be a BlobRecord type!")
 
-    async def transform_output(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
-    ) -> None:
-        blob_record = BlobRecord(data=await make_tar(runpath / location))
-        await transmitter.transmit_record(blob_record)
+    async def transform_output(self, mime: str, location: Path) -> Record:
+        return BlobRecord(data=await make_tar(location))
 
 
 class ExecutableRecordTransformation(RecordTransformation):
     async def transform_input(
-        self, transmitter: RecordTransmitter, mime: str, runpath: Path, location: Path
+        self, record: Record, mime: str, runpath: Path, location: Path
     ) -> None:
-        # pre-make bin folder if neccessary
+        # pre-make bin folder if necessary
         base_path = Path(runpath / _BIN_FOLDER)
         base_path.mkdir(parents=True, exist_ok=True)
 
         # create file(s)
         _prepare_location(base_path, location)
-        await transmitter.dump(base_path / location, mime)
+        await _save_record_to_file(record, base_path / location, mime)
```

Comment:
How `_save_record_to_file` will accomadate different transmitter dump functions?

Reply:
Yes, that optimization is lost. Right now the flow is load record into memory -> save to file. After this PR, the transformations should now nothing about the transmitters. That was the goal of the issue, but that will prevent some optimizations, we need see if those are acceptable sacrifices, or if we have to do something about it.

Reply:
I see now and it makes sense. But not sure that this will work with `RecordTree` since they require `_get_recordtree_transmitters` to be able to `load` -> `_load_record_from_file`.